gulp/plugin.js

```diff
@@ -1,340 +1,99 @@
 'use strict';
 
 var lazyReq = require('lazy-req')(require);
-var inquirer = lazyReq('inquirer');
-var through = lazyReq('through2');
 var path = lazyReq('path');
-var livereload = lazyReq('gulp-livereload');
-var watch = lazyReq('gulp-watch');
-var fs = require('fs-extra');
 var runSequence = require('run-sequence');
-var prettyTime = lazyReq('pretty-hrtime');
-
-var PLUGIN_PATHS = {
-  SCRIPTS: 'plugin/res/js/angularjs',
-  SCRIPT_DEPENDENCIES: 'plugin/res/js/angularjs/metadata',
-  TEXT: 'plugin/res/lang/feature'
-};
+var rsync = lazyReq('../lib/rsync.js');
+var extend = lazyReq('node.extend');
 
 module.exports = function (gulp, gutil) {
-  var scripts, sandboxApi, text, plugin, pluginUpload, pluginServer, lrListening, skins, server;
-
-  function getScripts() {
-    return scripts ? scripts : scripts = require('../lib/scripts.js')(gulp, gutil);
-  }
-
-  function getSandboxApi() {
-    return sandboxApi ? sandboxApi : sandboxApi = require('../lib/sandbox-api-hack.js')(gulp, gutil);
-  }
-
-  function getText() {
-    return text ? text : text = require('../lib/text.js')(gulp, gutil);;
-  }
-
-  function getPlugin() {
-    return plugin ? plugin : plugin = require('../lib/plugin-create.js')(gulp, gutil);
-  }
-
-  function getSkins() {
-    return skins ? skins : skins = require('../lib/skins.js')(gulp, gutil);
-  }
-
-  function getServer() {
-    return server ? server : server = require('../lib/server.js')(gulp, gutil);
-  }
+  var pluginUpload = require('../lib/plugin-upload.js')(gulp, gutil);
+  var sandboxApi = require('../lib/sandbox-api-hack.js')(gulp, gutil);
```

Comment:
This needs to be added in conditionally, or else you get an error when run project-specific SDK commands (like `li export-plugin`) like the following:
```
}LT5542:linc15 doug.schroeder$ li export-plugin
[11:06:24] A sandboxPluginDir is required in your configuration. Please use template.server.conf.json to create server.conf.json.
```